source/common/res/features/reports/main.js

```diff
@@ -1,549 +1,512 @@
-          var netWorths = ynabToolKit.reports.netWorth.netWorths;
-          var chart = ynabToolKit.reports.netWorthReportChart;
-          var values = [labels[0], labels[0]];
-
-          if ($('#reports-filter').is(':visible')) {
-              // Ok, the filter is in use, set the values to the filter values.
-            values = document.getElementById('reports-date-filter').noUiSlider.get();
+        // once the user changes the select find out if it's one of the "grouped" options (all/on/off) if it
+        // is, then get rid of the selected accounts array, if it's not then add it to the selected accounts array
+        $select.change(function () {
+          if (nonAccountOptions.indexOf($select.val()) !== -1) {
+            selectedAccounts = [];
+          } else if (selectedAccounts.indexOf(this.value) === -1) {
+            selectedAccounts.push(this.value);
           }
 
-          var startIndex = labels.indexOf(values[0]);
-          var endIndex = labels.indexOf(values[1]);
+          updateAccountList();
+          filterTransactionsAndBuildChart();
+        });
 
-            // Save the date filter values in case they go to another tab and come back.
-          sessionStorage.setItem('reportsDateFilter', values);
+        // this function updates the "chips" on the page every time the user changes the select
+        function updateAccountList() {
+          // remove them all first
+          $accountList.empty();
 
-          chart.data.labels = labels.slice(startIndex, endIndex + 1);
-          chart.data.datasets[0].data = liabilities.slice(startIndex, endIndex + 1);
-          chart.data.datasets[1].data = assets.slice(startIndex, endIndex + 1);
-          chart.data.datasets[2].data = netWorths.slice(startIndex, endIndex + 1);
+          if (selectedAccounts.length === 0) {
+            // if the selected accounts are empty and we didn't just click one of the "all" options
+            // then go ahead and set the select to whatever our default for the report is.
+            if (nonAccountOptions.indexOf($select.val()) === -1) {
+              $select.val(availableAccountTypes);
+            }
+          } else {
+            $select.val('custom');
+          }
 
-          chart.update();
+          // for each selected account, add a chip to the page when someone clicks the chip, it will get
+          // removed from the list of chips. if they've clicked the last chip then the above code will default
+          // us to whatever the default option of the report is
+          selectedAccounts.forEach((accountId) => {
+            let accountData = onBudgetAccounts.find((account) => accountId === account.get('entityId')) ||
+                              offBudgetAccounts.find((account) => accountId === account.get('entityId'));
+            $accountList
+              .append(
+                $(`<div class="account-chip" title="${accountData.get('accountName')}">${accountData.get('accountName')}</div>`)
```

Comment:
I'm fully aware that this would involve someone naming their account with `<script></script>` etc, but I still suspect this will get us knocked back in Firefox review. Can we update to the JQuery way of passing the object in with options so they get escaped?